spec/requests/create_dro_spec.rb

```diff
@@ -218,6 +219,56 @@
     end
   end
 
+  context 'when the file is application/x-stanford-json' do
+    let(:blob) do
+      ActiveStorage::Blob.create!(key: 'tozuehlw6e8du20vn1xfzmiifyok',
+                                  filename: 'file2.txt', content_type: 'application/x-stanford-json',
+                                  byte_size: 10, checksum:)
+    end
+    let(:expected_content_type) { 'application/json' }
+
+    it 'switches the content type to application/json' do
+      post '/v1/resources',
+           params: request,
+           headers: { 'Content-Type' => 'application/json', 'Authorization' => "Bearer #{jwt}" }
+      expect(response).to be_created
+      expect(response.location).to be_present
+      expect(JSON.parse(response.body)['jobId']).to be_present
+      expect(IngestJob).to have_received(:perform_later).with(model_params: expected_model_params,
+                                                              background_job_result: instance_of(BackgroundJobResult),
+                                                              signed_ids: { 'file2.txt' => signed_id },
+                                                              globus_ids: {},
+                                                              start_workflow: false,
+                                                              assign_doi: false,
+                                                              priority: 'default')
+    end
+  end
+
+  context 'when the file supplies a nil content_type' do
+    let(:blob) do
+      ActiveStorage::Blob.create!(key: 'tozuehlw6e8du20vn1xfzmiifyok',
+                                  filename: 'file2.txt', content_type: nil,
+                                  byte_size: 10, checksum:)
+    end
+    let(:expected_content_type) { 'application/octet-stream' }
+
+    it 'switches the content type to application/octet-stream' do
+      post '/v1/resources',
+           params: request,
+           headers: { 'Content-Type' => 'application/json', 'Authorization' => "Bearer #{jwt}" }
```

Comment:
would it be more like actual usage to omit the `Content-Type` header here?

Reply:
as above, I think this is the content type of the wrapper, the blob is the one missing the content type

Reply:
for example, see https://github.com/sul-dlss/sdr-client/blob/main/lib/sdr_client/deposit/upload_files.rb#L51 and https://github.com/sul-dlss/sdr-client/blob/main/lib/sdr_client/deposit/create_resource.rb#L48-L51 for where we post to sdr-api with `application/json` type as the header (with the body being a blob with the separate content-type we are munging here)

Reply:
yup, i think you're right, thanks for the explanation!